presto-spi/src/main/java/com/facebook/presto/spi/block/ByteArrayBlockBuilder.java

```diff
@@ -119,7 +124,7 @@ private void growCapacity()
 
     private void updateDataSize()
     {
-        retainedSizeInBytes = INSTANCE_SIZE + sizeOf(valueIsNull) + sizeOf(values);
+        retainedSizeInBytes = INSTANCE_SIZE + sizeOf(valueIsNull) + sizeOf(values) + NULL_VALUE_BLOCK.getRetainedSizeInBytes();
```

Comment:
This is a constant shared across all instances, so should not be part of the retained size.

Reply:
Okay. But then we had a issue with the
Test code that deals with retained size. It also references the static field and throws an error saying that retainedSize didn’t match. So added this constant. The workaround is add these NULL_VALUE_BLOCK in BlockUtil and refer it from there.

Reply:
> Test code that deals with retained size. It also references the static field 

Maybe it should skip static fields?

Reply:
Yeah. I will change the test code to skip static fields.